Data Science Fundamentals: Python |
[Table of Contents](../index.ipynb)
- - - 
<!--NAVIGATION-->
Module 1. [Basic Python Syntax](./01_mod_python_syntax.ipynb) | [Variables and Objects](./02_python_variables_objects.ipynb) | [Operators](./03_python_operators.ipynb) | **[Exercises](./04_mod_practice.ipynb)**

# Module 1: Practice Exercises
Please remember to [Submit this file to Titus](https://titus.techtalentsouth.com/mod/assign/view.php?id=44970)

# Exercise 1: Accept the user's first and last name and print them in reverse order with a space between them.

In [5]:
### Accept user's first and last name 
firstname, lastname = input("Please type your first and last name with a space between here: ").split(" ")

### Print last before first name
print(lastname, firstname)

Please type your first and last name with a space between here: Tom Jones
Jones Tom


### Exercise 2. Accept an integer (n) and compute the value of n+nn+nnn

In [2]:
### Accept an interger "n" from user
n = int(input("Please enter a number between 1 and 9: "))

### Compute input value for n as n+nn+nnn
print(123*n)

Please enter a number between 1 and 9: 2
246


### Exercise 3. Ask the user "What country are you from?" then print the following statement: "I have heard that [input] is a beautiful country!"

In [3]:
### Ask the user "What country are you from?"
country = (input("What country are you from?: "))

### Print the statement "I have heard that [input] is a beautiful country!"
print("I have heard that " + country + " is a beautiful country!")

What country are you from?: Brazil
I have heard that Brazil is a beautiful country!


### Exercise 4. What is the output of the following python code?

In [12]:
# set x value
x = 10

# set y value
y = 50

# create if statement for x**2>100 and y<100
if (x ** 2 > 100 and y < 100):
    
# print x and y    
    print(x, y)

The output is nothing. `X**2` is not greater than 100. The if statement is false

### Exercise 5: What is the output of the following addition `+` operator, and why does this code chunk execute this way??

In [8]:
#Print your answer here
a = [10, 20]

#Set the variable "b" to point to variable "a"
b = a

#add the additional values of "30" and "40" to b
b += [30, 40]

#print a
print(a)

#print b
print(b)

[10, 20, 30, 40]
[10, 20, 30, 40]


Both a and b are variable names that point to lists. The operator "+=" is an addition assinment operator and in this case it's adding values to the lists b and thusly a, as members of the list in the order they were presented. 

### Exercise 6. What is the output of the following code, and what operator is being used here?

In [9]:
#Print(2%6)
print(2%6)

2


This operator is the modulus operator. This operator takes a x and a y value and divides x by y, only presenting the remainder. 2 is not divisible by 6 without reducing it to a demical so the remainder is 2.

### Exercise 7: What is the output of the following code, and what arithmetic operators are used in this code?

In [10]:
#print(2 * 3 ** 3 * 4)
print(2 * 3 ** 3 * 4)

216


The output is 216. In this code, the opertators of multiplaction and exponentiation are used.

### Exercise 8: What is a text editor?

Text editors are programs that allow users to type, edit, and save files as plain text files. Different from a word processor as there is no formatting with a text editor 

### Exercise 9: What is python?

Python is a object-oriented, dynamically typed, psuedo-code language. 

### Exercise 10: What is jupyter notebook, what type of python environment is it, and what alternatives are there to jupyter notebook?

Jupyter Notebooks is a web-based application with an IPython environment for coding in python. PyCharm, Rodeo IDE, and Codenvy are a few examples of alternatives for Jupyter Notebooks.

### Please remember to submit this file to Titus!

<!--NAVIGATION-->
Module 1. [Basic Python Syntax](./01_mod_python_syntax.ipynb) | [Variables and Objects](./02_python_variables_objects.ipynb) | [Operators](./03_python_operators.ipynb) | **[Practice](./04_mod_practice.ipynb)**
<br>
[top](#)

- - -

Copyright © 2020 Qualex Consulting Services Incorporated.